velox/functions/lib/Re2Functions.cpp

```diff
@@ -125,18 +125,21 @@ bool re2Extract(
   }
 }
 
-std::string
-likePatternToRe2(StringView pattern, char escapeChar, bool& validPattern) {
+std::string likePatternToRe2(
+    StringView pattern,
+    bool hasEscape,
+    char escapeChar,
+    bool& validPattern) {
   std::string regex;
   validPattern = true;
   regex.reserve(pattern.size() * 2);
   regex.append("^");
   bool escaped = false;
   for (const char c : pattern) {
-    if (escaped && !(c == '%' || c == '_' || c == escapeChar)) {
+    if (hasEscape && escaped && !(c == '%' || c == '_' || c == escapeChar)) {
```

Comment:
If you used a `std::optional` for `escapeChar`, no changes would be needed to the body of this function. In particular these conditions would be noticeably simpler IMO.